### PR TITLE
Improvements to error handling for support ticket configuration

### LINKED ISSUE #2867: support ticket crashes if misconfigured
From discourse - below - the `support_ticket` feature seems to crash if it's misconfigured.

https://discourse.openondemand.org/t/support-ticket-is-not-working/2733

### PR DESCRIPTION
**Pull request by abujeda**
Fixes #2867 

Improvements to error handling for support ticket configuration in controller

### CODE REVIEW

File: apps/dashboard/app/controllers/support_ticket_controller.rb [hunk 1]
```diff
@@ -17,6 +17,10 @@ def new
     @support_ticket = support_service.default_support_ticket(params)
 
     render get_ui_template
+
+  rescue StandardError => e
+    logger.error "Could not render support ticket page. Error=#{e}"
+    redirect_to root_url, :flash => { :alert => t('dashboard.support_ticket.generic_error', error: e) }
```
Comment (johrstrom) [inline apps/dashboard/app/controllers/support_ticket_controller.rb:23]: Can this be a more specific error? The idea I had in #2992 was to say specifically it's misconfigured. I think that's what we need (along with the redirect) is a specific message saying it's not configured correctly.

Comment (abujeda) [reply apps/dashboard/app/controllers/support_ticket_controller.rb:23]: will update

Comment (abujeda) [reply apps/dashboard/app/controllers/support_ticket_controller.rb:23]: Fixed. Let me know if this is what you had in mind

Comment (johrstrom) [reply apps/dashboard/app/controllers/support_ticket_controller.rb:23]: Ummm, not quite. I had 

`"support_ticket is misconfigured. 'email' or 'rt_api' sections are required."`
You may even go further with 
`"support_ticket is misconfigured. 'email' or 'rt_api' sections are required in the configuration YAML."`

`service class` is a bit esoteric. I could be persuaded into even including a documentation link. The gist is basically - we have to be super explicit and leave nothing to interpretation. And **even then** we'll get tickets for the same, but maybe not as many (🤞).

Comment (abujeda) [reply apps/dashboard/app/controllers/support_ticket_controller.rb:23]: (jaja) updated again. Added a more specific error message that will be shown to the user

File: apps/dashboard/app/models/user_configuration.rb [hunk 2]
```diff
@@ -179,4 +179,6 @@ def add_property_methods
   end
 
   attr_reader :config, :request_hostname
+
+  class UserConfigurationError < StandardError; end
```
Comment (johrstrom) [inline apps/dashboard/app/models/user_configuration.rb:183]: Looks like we don't need this anymore.

File: apps/dashboard/app/controllers/support_ticket_controller.rb (hunk 1 above)
Comment (johrstrom) [reply apps/dashboard/app/controllers/support_ticket_controller.rb:23]: That'll work just fine - thank you so much!

File: apps/dashboard/app/models/user_configuration.rb (hunk 2 above)
Comment (abujeda) [reply apps/dashboard/app/models/user_configuration.rb]: Sorry - yes, removed

**Review by johrstrom — APPROVED** [review]
Thanks!

**Pull request merged**